test/test_performance_specializations.jl

```diff
@@ -0,0 +1,56 @@
+module TestPerformanceSpecializations
+
+using Test
+using Trixi
+
+include("test_trixi.jl")
+
+# Start with a clean environment: remove Trixi output directory if it exists
+outdir = "out"
+isdir(outdir) && rm(outdir, recursive=true)
+
+
+@testset "Performance specializations" begin
+  @timed_testset "TreeMesh3D, flux_shima_etal" begin
+    trixi_include(@__MODULE__,
+      joinpath(examples_dir(), "tree_3d_dgsem", "elixir_euler_ec.jl"),
+      initial_refinement_level=0, tspan=(0.0, 0.0),
+      volume_flux=flux_shima_etal, surface_flux=flux_shima_etal)
+    u_ode = copy(sol.u[end])
+    du_ode = zero(u_ode)
+
+    GC.@preserve u_ode du_ode begin
```

Comment:
```suggestion
    # Preserve original memory since it will be `unsafe_wrap`ped and might thus otherwise be garbage collected
    GC.@preserve u_ode du_ode begin
```
Maybe for future me 😬  (feel free to modify if incorrect)?